proof_ics23.go

```diff
@@ -179,16 +179,12 @@ func convertVarIntToBytes(orig int64, buf [binary.MaxVarintLen64]byte) []byte {
 	return buf[:n]
 }
 
-// GetWithProof gets the proof for the given key.
-func (t *ImmutableTree) GetWithProof(key []byte) (*ics23.CommitmentProof, error) {
+// GetProof gets the proof for the given key.
+func (t *ImmutableTree) GetProof(key []byte, isExist bool) (*ics23.CommitmentProof, error) {
 	if t.root == nil {
 		return nil, fmt.Errorf("cannot generate the proof with nil root")
 	}
-	value, err := t.Get(key)
-	if err != nil {
-		return nil, err
-	}
-	if value != nil {
+	if isExist {
```

Comment:
sorry for not being more specific, I think you can use the [`Has`](https://github.com/cosmos/iavl/blob/master/immutable_tree.go#L143) function

```go
exist, err := t.Has(key)
if err != nil {
    return nil, err
}

if exist {
    return t.GetMembershipProof(key)
}

return t.GetNonMembershipProof(key)
```

Reply:
I guess when the SDK queries for this value, it should be performing the `Has` query then?